### PR TITLE
FIX-#4498: Pin `protobuf<4.0` to successfully build docs

### LINKED ISSUE #4498: CI Fails on Build Docs
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Modin version** (`modin.__version__`):
- **Python version**:
- **Code we can use to reproduce**:

### Describe the problem

Seems that a new major version of protobuf has been released, and our CI automatically downloads it. The ray project has old generated protos and some things are now disallowed/changed in the new release, so our `build_docs` fails. We need to pin protobuf to `<=3.20` until the ray team regens their protos.

### PR DESCRIPTION
**Pull request by RehanSD**
Signed-off-by: Rehan Durrani <rehan@ponder.io>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4498 ? 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: docs/requirements-doc.txt [hunk 1]
```diff
@@ -13,3 +13,5 @@ git+https://github.com/modin-project/modin.git@master#egg=modin[all]
 sphinxcontrib_plantuml
 sphinx-issues
 xgboost
+# TODO: remove protobuf dependency when ray-project resolves GH#252025
```
Comment (YarShev) [inline docs/requirements-doc.txt:16]:
```suggestion
# TODO: remove protobuf dependency when ray-project resolves GH#25205
```

File: setup.py [hunk 2]
```diff
@@ -6,7 +6,8 @@
 
 dask_deps = ["dask>=2.22.0,<2022.2.0", "distributed>=2.22.0,<2022.2.0"]
 # TODO: remove redis dependency when resolving GH#4398
-ray_deps = ["ray[default]>=1.4.0", "pyarrow>=4.0.1", "redis>=3.5.0,<4.0.0"]
+# TODO: remove protobuf dependency when ray-project resolves GH#252025
```
Comment (YarShev) [inline setup.py:9]:
```suggestion
# TODO: remove protobuf dependency when ray-project resolves GH#25205
```

**Review by YarShev — COMMENTED** [review]
Should we also add a pin for protobuf in other env files?

File: docs/requirements-doc.txt [hunk 3]
```diff
@@ -13,3 +13,5 @@ git+https://github.com/modin-project/modin.git@master#egg=modin[all]
 sphinxcontrib_plantuml
 sphinx-issues
 xgboost
+# TODO: remove protobuf dependency when ray-project resolves GH#25205
+protobuf<4.0
```
Comment (YarShev) [inline docs/requirements-doc.txt:17]: Is this required here? We install `git+https://github.com/modin-project/modin.git@master#egg=modin[all]` above so a pin for protobuf in setup.py should be enough.

File: environment-dev.yml [hunk 4]
```diff
@@ -45,5 +45,7 @@ dependencies:
       - connectorx>=0.2.6a4
       # TODO: remove when resolving GH#4398
       - redis>=3.5.0,<4.0.0
+      # TODO: remove when ray-project resolves GH#25205
+      - protobuf<4.0
```
[the remark this answers is not in the document]
Comment (RehanSD) [reply environment-dev.yml:49]: I don't think so - this only breaks when we install ray, so if someone is installing without an engine, I don't think it's a huge concern.

File: docs/requirements-doc.txt (hunk 3 above)
Comment (RehanSD) [reply docs/requirements-doc.txt]: Removed!

File: environment-dev.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (YarShev) [reply environment-dev.yml:49]: We install Ray with deps from `requirements/requirements-no-engine.yml` for `test-ray-master` job. We should probably add `pip install protobuf<4.0` around here. https://github.com/modin-project/modin/blob/60518ca737bf13cc37c8f1269985dcbb68dad4c7/.github/workflows/push-to-master.yml#L24

[the remark this answers is not in the document]
Comment (RehanSD) [reply environment-dev.yml:49]: Makes sense - will add!

File: docs/requirements-doc.txt (hunk 3 above)
Comment (YarShev) [reply docs/requirements-doc.txt:17]: It is strange that protobuf > 4.0 is installed for build docs.

**Review by devin-petersohn — COMMENTED** [review]
protobuf 4+ was yanked from pypi. I think this can be closed without merging

**Comment by devin-petersohn** [comment]
Link: https://pypi.org/project/protobuf/#history

**Comment by RehanSD** [comment]
Given that protobuf has been yanked, I agree - lets close this!

**Pull request closed without merging**